os/drivers/audio/alc5658.c

```diff
@@ -615,15 +617,19 @@ static int alc5658_getcaps(FAR struct audio_lowerhalf_s *dev, int type, FAR stru
 
 		switch (caps->ac_format.hw) {
 		case AUDIO_FU_VOLUME:
+#ifndef CONFIG_AUDIO_EXCLUDE_VOLUME
 			caps->ac_controls.hw[0] = ALC5658_HP_VOL_MAX;
 			caps->ac_controls.hw[1] = priv->volume;
 			break;
+#endif
 		case AUDIO_FU_INP_GAIN:
+#ifndef CONFIG_AUDIO_EXCLUDE_GAIN
 			caps->ac_controls.hw[0] = ALC5658_GAIN_MAX;
 			caps->ac_controls.hw[1] = priv->gain;
 			break;
+#endif
 		default:
-			break;
+			return -ENOSYS;
```

Comment:
Now we added -ENOSYS if device doesn't support volume control.
But get_audio_volume still checking only below.
```
	ret = ioctl(fd, AUDIOIOC_GETCAPS, (unsigned long *)&caps_desc.caps);
	if (ret < 0) {
		meddbg("Fail to ioctl AUDIOIOC_GETCAPS\n");
		return AUDIO_MANAGER_DEVICE_FAIL;
	}
```
This means if we enable this config, volume control will return same error value, AUDIO_MANAGER_DEVICE_FAIL